solr/core/src/java/org/apache/solr/request/SimpleFacets.java

```diff
@@ -903,7 +910,7 @@ public void execute(Runnable r) {
 
   private int numDocs(String term, final SchemaField sf, final FieldType ft, final DocSet baseDocset) {
     try {
-      return searcher.numDocs(ft.getFieldQuery(null, sf, term), baseDocset);
+      return baseDocset.size() == 0? 0: searcher.numDocs(ft.getFieldQuery(null, sf, term), baseDocset);
```

Comment:
Line 903 (immediately above) is the only caller of this private method. Would it make sense to do the check in only one of these places? Maybe prefer the higher-level (line 903) check, where the check could also be used to bypass `termCountEntries.sorted(...)`? Could optionally leave `assert baseDocset.size() > 0` here.

Reply:
sort by count won't be done if baseDocSet size is 0 but I have kept this check in numDocs so that, any future usage can benefit from it